wp-content/plugins/tapestry/controller/class.user-controller.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Add/update/retrieve User progress
+ * 
+ */
+class TapestryUserController {
+
+    /** 
+     * Update User's video progress for a tapestry post
+     * @param type @userId the user's id
+     * @param type @postId the post's ID
+     * @param type @progressData json data for progress of each node
+    */
+    public function updateProgress($userId, $postId, $progressData) {
```

Comment:
Rather than updating the whole progress data, I think we should store the progress _per node_. We'll be doing this save very often, therefore it would be much faster if the data load is smaller.

Reply:
Saving node progress one node at a time